component/core/src/main/java/org/exoplatform/social/core/metadata/favorite/FavoriteServiceImpl.java

```diff
@@ -39,15 +40,13 @@ public FavoriteServiceImpl(MetadataService metadataService) {
   }
 
   @Override
-  public void createFavorite(Favorite favorite) throws ObjectAlreadyExistsException {
+  public void createFavorite(Favorite favorite, Map<String, String> favoriteProperties) throws ObjectAlreadyExistsException {
```

Comment:
```suggestion
  @Override
  public void createFavorite(Favorite favorite) throws ObjectAlreadyExistsException {
    this.createFavorite(favorite, false);
  }

  @Override
  public void createFavorite(Favorite favorite, boolean hiddeInList) throws ObjectAlreadyExistsException {
```
In term of API definition, multiple things:
1/ It's better to keep the original operation with simple input without options `createFavorite(Favorite favorite)`. This will allow to use the API for simple use cases without complexity.
2/ It's better to not add a generic `properties` Map while our use case is simple: whether to hide favorites while listing or not.
3/ You added an option for favorite creation, but it's not clear (in term of API perspective) where it will be used. In fact, a new method to list favorites should be added in Service layer to allow filter by visibility.